optaplanner-examples/src/main/java/org/optaplanner/examples/taskassigning_listvariable/swingui/TaskAssigningPanel.java

```diff
@@ -135,7 +135,7 @@ public void consumeUpTo(final int consumedTime) {
             solution.setFrozenCutoff(consumedTime);
             for (Task task : solution.getTaskList()) {
                 if (!task.isPinned()) {
-                    if (task.getStartTime() != null && task.getStartTime() < consumedTime) {
+                    if (task.getEmployee() != null && task.getStartTime() < consumedTime) {
```

Comment:
shouldn't we do the startTime null check anyway? True, if employee is not null, start time won't be null either, but it seems a bit brittle for future refactors?

Reply:
`task.employee != null => task.startTime != null` holds but `task.employee == null => task.startTime == null` does not.

It's because the start time updating listener is not perfect and does not unset the start time before a variable changes.

So, when you manually unassign a task, its `employee` becomes `null` but its `startTime` remains non-null.

We could fix the listener to do that but I'm not sure whether the increased complexity is appropriate for the example. But I guess it could be. Correctness is worth any amount of complexity, especially for code that we use to teach people how to do it correctly.

Without that, looking at `task.employee` is currently the safe way to tell whether the task is assigned or unassigned.

Reply:
Let's not introduce unsetting of `startTime` in `beforeVariableChanged()` for now. I don't want that to skew benchmark results if we compare this to the original chained model.